src/gui/widgets/text_box_base.cpp

```diff
@@ -660,15 +660,18 @@ void text_box_base::signal_handler_sdl_key_down(const event::ui_event event,
 
 		case SDLK_RETURN:
 		case SDLK_KP_ENTER:
-			if ((modifier & KMOD_SHIFT) && is_editable()) {
-				insert_char("\n");
-			}
 
-			if(!is_composing() || (modifier & (KMOD_CTRL | KMOD_ALT | KMOD_GUI))) {
-				return;
+//			if(!is_composing()) {
+//				return;
+//			}
+
+			// If pressed with modifier, let the subclass handle it
```

Comment:
I don't think there's a reason to require a modifier here? Doesn't the "default button" functionality prevent this point from being reached? If that functionality is disabled in a dialog, I think you should be able to press enter without a modifier to insert a newline.

Also, is it really okay to remove the composing check? I'll admit I don't entirely understand what that does, but… I assume it affects typing with dead keys or input methods (eg Japanese, Chinese).

Reply:
Pressing Enter now works, but required a different fix. Please check 70691ad648cabf9fb340968e3642da7e2198ccff.
Currently, the window checks if the widget with keyboard focus is a multiline_text or not, and trigger default behavior in the latter case, and for the first it passed down control.

Reply:
I see that.

I don't mind plain Enter not working in most dialogs; I just think it should at least work in a dialog that turns off the behaviour of closing on Enter.